### ISSUE TITLE
[v2.3.1] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#125425]
### 🐛 Describe the bug

This issue is for tracking cherry-picks to the release branch. Following is [release branch](https://github.com/pytorch/pytorch/tree/release/2.3) for the 2.3.1 release.

Our plan from this point is roughly the following:

* Phase 1 (until 5/27): Cherry-pick post deadline (End of day 5PM PST)
* Phase 2 (after 5/27):  Perform extended integration/stability/performance testing based on Release Candidate builds.

**Only issues that have ‘cherry-picks’ in this tracker will be considered for the release.**

## Cherry-Pick Criteria

**Phase 1 (until 5/27):**

The Releng team relies on the cherry pick process to manage risk to release quality, i.e. by porting a small set of commit from trunk that are "must-have" into the release branch, we limit the change to the minimal to address pressing issues. Thus, not everything a developer land into the trunk will make it into the release. So, please consider the criteria below and follow the cherry picking process. Only low-risk changes may be cherry-picked from master:

1. No feature work allowed
2. Fixes to regressions against the most recent release (e.g. 2.3.0 for 2.3.1 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
3. Low risk critical fixes for: silent correctness, backwards compatibility, crashes, deadlocks, (large) memory leaks
4. Critical Fixes to new features being introduced in 2.3.0 release
5. Documentation improvements
6. Release branch specific changes (e.g. blocking ci fixes, change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @huydhn, @PaliC,  @malfet). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain. 

**Phase 2 (after 5/27):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.1).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.3
    git checkout release/2.3
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/2.3' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.3
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed trunk PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

See [HUD 2.3](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.3/1?per_page=50) 

### Versions

2.3.1

**Comment by snadampal** [comment]
Link to main PR (if applicable):
- https://github.com/pytorch/builder/pull/1803

Link to 2.3.1 release branch PR:
- https://github.com/pytorch/builder/pull/1805

Criteria Category:
Fixes performance regression issues https://github.com/pytorch/builder/issues/1774 and https://github.com/pytorch/pytorch/issues/124922

**Comment by saitcakmak** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/125265

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/125443

Criteria Category:
* Documentation improvements

**Comment by soulitzer** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/124634

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/125755

Criteria Category:
* low risk critical fix

**Comment by eqy** [comment]
Link to landed trunk PR (if applicable):

- This is not a PR cherrypick but a revert commit: https://github.com/pytorch/pytorch/commit/51cf57c6c644e6ab2104d7f9744b3722bd82d665

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/125790

Criteria Category:

- Low risk critical fix

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/122616

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/126094

Criteria Category:
* Critical Fix - torchdata compatibility

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/125684

Link to release branch PR:
* 

Criteria Category:
* Critical fix for https://github.com/pytorch/pytorch/issues/125109

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/125617

Link to release branch PR:
* 

Criteria Category:
* Critical fix for: https://github.com/pytorch/pytorch/issues/125094

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/122895

Link to release branch PR:
* 

Criteria Category:
* Fixes doc failure on Release branch

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/125355

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/126100

Criteria Category:
* Critical Fixes https://github.com/pytorch/pytorch/issues/124850

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/125154

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/126101

Criteria Category:
* Fixes https://github.com/pytorch/pytorch/issues/124868

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/124635

Link to release branch PR:
* 

Criteria Category:
* Critical fix https://github.com/pytorch/pytorch/issues/124335

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/124238

Link to release branch PR:
* 

Criteria Category:
* Critical fix https://github.com/pytorch/pytorch/issues/119607

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/123407

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/123482

Criteria Category:
* Fixes new feature

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/125843

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/126108

Criteria Category:
* Required to fix Lint errors on release

**Comment by wanchaol** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/124780

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/126147

Criteria Category:
* Fixes to regressions against the 2.3.0 release

**Comment by antoinebrl** [comment]
Hello 👋!
I am facing some issue regarding none-persistent buffers in distributed scenarios. A solution was submitted and merged into `main` (https://github.com/pytorch/pytorch/pull/125337). I was doing the cherry picking process to submit this fix for it to be part of the next patch release. However, the solution was developed after a refactoring which tackled another problem introduced in 2.3 (https://github.com/pytorch/pytorch/commit/a89177936cc46c476f73a555f499a89ee4c3aa01). This create a conflict preventing me to submit the cherry picked commit as is. Should I submit both modifications in the corresponding order for them to be included in 2.3.1?
Tagging @fegin who authored both contributions.

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/124932

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/126301

Criteria Category:
* Documentation

**Comment by williamwen42** [comment]
Link to landed trunk PR (if applicable):

https://github.com/pytorch/pytorch/pull/120756

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/126332

Criteria Category:

Critical fix https://github.com/pytorch/pytorch/issues/119607 torch.compile refleak

**Comment by jithunnair-amd** [comment]
Link to landed trunk PR (if applicable):
- https://github.com/pytorch/pytorch/pull/121030

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/126448

Criteria Category:
- Low risk addition of hipify mappings to enable DeepSpeed transformer extensions on ROCm

**Comment by weifengpy** [comment]
Link to landed trunk PR (if applicable):
https://github.com/pytorch/pytorch/pull/123617

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/126464

Criteria Category:
Fixes to regressions against the 2.3.0 release

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/126452

Criteria Category:
* Release only changes - pin docker image for rocm CI

**Comment by mvpatel2000** [comment]
Link to main PR (if applicable):
- https://github.com/pytorch/pytorch/pull/121544

Link to 2.3.1 release branch PR:
- https://github.com/pytorch/pytorch/pull/126557

Criteria Category:
- Low risk critical fix for checkpointing. This PR removes an additional check introduced in torch 2.3 which is actually incorrect and causes checkpointing to fail if at least 1 forward/backward pass has not been run.

**Comment by mvpatel2000** [comment]
Link to main PR (if applicable):
- https://github.com/pytorch/pytorch/pull/124698

Link to 2.3.1 release branch PR:
- https://github.com/pytorch/pytorch/pull/126559

Criteria Category:
- Low risk critical fix for checkpointing. When checkpointing with activation checkpointing, the names of the variables (FQNs) are changed with the tag `_checkpoint_wrapped_module`. This breaks checkpointing if you resume without activation checkpointing.

**Comment by mvpatel2000** [comment]
Link to main PR:
- https://github.com/pytorch/pytorch/pull/125336

Link to 2.3.1 release branch PR: 
- https://github.com/pytorch/pytorch/pull/126567

Criteria Category:
- Low risk critical fix for checkpointing. Torch 2.3 ignores `_extra_state` (whereas prior PyTorch versions correctly handle it), breaking checkpoint loading from prior checkpoints. This also breaks integration with Nvidia's TransformerEngine.

**Comment by antoinebrl** [comment]
⚠️ https://github.com/pytorch/pytorch/pull/126567 must be merged before

Link to landed trunk PR:
- https://github.com/pytorch/pytorch/pull/125337

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/127219

Criteria Category:
- Low risk critical fix for checkpointing error introduced in 2.3. Fixes https://github.com/pytorch/pytorch/issues/122792

**Comment by atalman** [comment]
Please note, we are in the phase:
Phase 2 (after 5/27): Perform extended integration/stability/performance testing based on Release Candidate builds.
We do not accept any new cherry-picks

**Comment by huydhn** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/126414

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/127312

Criteria Category:
* CI fix to keep inductor job green in release branch

**Comment by atalman** [comment]
closing release tracker. release published
